contracts/anchor_basset_hub/src/state.rs

```diff
@@ -245,25 +250,53 @@ type OldUnbondWaitList = (Vec<u8>, Uint128);
 
 pub fn read_old_unbond_wait_lists(
     storage: &mut dyn Storage,
+    limit: Option<u32>,
 ) -> StdResult<Vec<StdResult<OldUnbondWaitList>>> {
-    let reader: ReadonlyBucket<Uint128> = ReadonlyBucket::multilevel(storage, &[PREFIX_WAIT_MAP]);
+    let reader: ReadonlyBucket<Uint128> =
+        ReadonlyBucket::multilevel(storage, &[OLD_PREFIX_WAIT_MAP]);
     Ok(reader
         .range(None, None, Order::Ascending)
+        .take(limit.unwrap_or(MAX_DEFAULT_RANGE_LIMIT) as usize)
         .collect::<Vec<StdResult<OldUnbondWaitList>>>())
 }
 
-pub fn migrate_unbond_wait_lists(storage: &mut dyn Storage) -> StdResult<()> {
-    let old_unbond_wait_list = read_old_unbond_wait_lists(storage)?;
-    let mut bucket: Bucket<UnbondWaitEntity> = Bucket::multilevel(storage, &[PREFIX_WAIT_MAP]);
-    for res in old_unbond_wait_list {
-        let (key, amount) = res?;
-        let unbond_wait_entity = UnbondWaitEntity {
-            bluna_amount: amount,
-            stluna_amount: Uint128::zero(),
-        };
-        bucket.save(&key, &unbond_wait_entity)?;
+// migrate_unbond_wait_lists moves the old values (Uint128) in OLD_PREFIX_WAIT_MAP storage to UnbondWaitEntity
+// in NEW_PREFIX_WAIT_MAP and deletes the old entries.
+pub fn migrate_unbond_wait_lists(
+    storage: &mut dyn Storage,
+    limit: Option<u32>,
+) -> StdResult<Response> {
+    let (removed_keys, num_migrated_entries) = {
+        let old_unbond_wait_list_entries = read_old_unbond_wait_lists(storage, limit)?;
+        let mut num_migrated_entries: u32 = 0;
+        let mut new_unbond_wait_list: Bucket<UnbondWaitEntity> =
+            Bucket::multilevel(storage, &[NEW_PREFIX_WAIT_MAP]);
+        let mut removed_keys: Vec<Vec<u8>> = vec![];
+
+        for res in old_unbond_wait_list_entries {
+            let (key, amount) = res?;
+            let unbond_wait_entity = UnbondWaitEntity {
+                bluna_amount: amount,
+                stluna_amount: Uint128::zero(),
+            };
+            new_unbond_wait_list.save(&key, &unbond_wait_entity)?;
+            removed_keys.push(key);
+            num_migrated_entries += 1;
+        }
+
+        (removed_keys, num_migrated_entries)
+    };
+
+    let mut old_unbond_wait_list: Bucket<Uint128> =
+        Bucket::multilevel(storage, &[OLD_PREFIX_WAIT_MAP]);
+    for key in removed_keys {
+        old_unbond_wait_list.remove(&key);
     }
-    Ok(())
+
+    Ok(Response::new().add_attributes(vec![
+        attr("action", "migrate_unbond_wait_lists"),
+        attr("num_migrated_entries", num_migrated_entries.to_string()),
+    ]))
```

Comment:
Looks good to me, just add to the test case a quick verification that the keys have actually been removed from the old unbond wait list.